docs/resources/pumls/stream.puml

```diff
@@ -0,0 +1,62 @@
+@startuml Stream
+
+!includeurl ./style.puml
+
+package "Stream" {
+
+    abstract class StreamProvider<<abstract>> {
```

Comment:
Add `__init__(name: str, pathname: str)` and `__iter__(): Iterator[Tuple[str, Any]]` functions